File: unleash/src/main/java/com/silvercar/unleash/UnleashException.java
package com.silvercar.unleash;

import com.silvercar.unleash.event.UnleashEvent;
import com.silvercar.unleash.event.UnleashSubscriber;

public class UnleashException extends RuntimeException implements UnleashEvent {

    public UnleashException(String message, Throwable cause) {
        super(message, cause);
    }

    @Override
    public void publishTo(UnleashSubscriber unleashSubscriber) {
        unleashSubscriber.onError(this);
    }
}

File: unleash/src/main/java/com/silvercar/unleash/UnleashException.kt
package com.silvercar.unleash

import com.silvercar.unleash.event.UnleashEvent
import com.silvercar.unleash.event.UnleashSubscriber

class UnleashException(message: String?, cause: Throwable?) :
  RuntimeException(message, cause), UnleashEvent {
  override fun publishTo(unleashSubscriber: UnleashSubscriber) {
    unleashSubscriber.onError(this)
  }
}
